be/src/vec/functions/function_string.h

```diff
@@ -4049,4 +4097,71 @@ class FunctionNgramSearch : public IFunction {
     }
 };
 
+class FunctionTranslate : public IFunction {
+public:
+    static constexpr auto name = "translate";
+    static FunctionPtr create() { return std::make_shared<FunctionTranslate>(); }
+    String get_name() const override { return name; }
+    size_t get_number_of_arguments() const override { return 3; }
+
+    DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
+        return std::make_shared<DataTypeString>();
+    };
+
+    DataTypes get_variadic_argument_types_impl() const override {
+        return {std::make_shared<DataTypeString>(), std::make_shared<DataTypeString>(),
+                std::make_shared<DataTypeString>()};
+    }
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        size_t result, size_t input_rows_count) const override {
+        // We need a local variable to hold a reference to the converted column.
+        // So that the converted column will not be released before we use it.
+        auto col_source =
+                block.get_by_position(arguments[0]).column->convert_to_full_column_if_const();
```

Comment:
better not use this. see what we deal the function with parameter columns in https://selectdb.feishu.cn/wiki/Rfwdw6VZCiQbAckxFHDcpKqknUf?from=from_copylink

Reply:
fixed, please review the code again🤓